pkg/patterns/addon/pkg/loaders/git.go

```diff
@@ -122,3 +142,72 @@ func parseGitURL(url string) GitRepository {
 		subDir:  subdir,
 	}
 }
+
+func handleExistingRepo(path string, auth transport.AuthMethod) error {
+	gitRepo, err := git.PlainOpen(path)
+	if err != nil {
+		return err
+	}
+
+	remote, err := gitRepo.Remote("origin")
+	if err != nil {
+		return err
+	}
+
+	err = remote.Fetch(&git.FetchOptions{
+		Force: true,
+		Auth:  auth,
+	})
+	if err != nil && err != git.NoErrAlreadyUpToDate {
+		return err
+	}
+
+	w, err := gitRepo.Worktree()
+	if err != nil {
+		return err
+	}
+
+	err = w.Checkout(&git.CheckoutOptions{
+		Branch: "refs/heads/master",
+	})
+	if err != nil {
+		return err
+	}
+
+	err = w.Reset(&git.ResetOptions{
+		Mode: git.HardReset,
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getAuthMethod() (transport.AuthMethod, error) {
+	sshFile := fmt.Sprintf("%s/.ssh/id_rsa", os.Getenv("HOME"))
```

Comment:
In the future, we could also support setting this directly in an env var (e.g. `GIT_SSH_KEY` or something like that).  But this is good

Reply:
That sounds great! I will do so in a follow-up PR.